dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ExecutorServiceImpl.java

```diff
@@ -673,14 +691,80 @@ private int createComplementCommandList(Date start, Date end, RunMode runMode, C
                         cmdParam.put(CMDPARAM_COMPLEMENT_DATA_END_DATE, DateUtils.dateToString(listDate.get(endDateIndex)));
                         command.setCommandParam(JSONUtils.toJsonString(cmdParam));
                         processService.createCommand(command);
+
+                        if (schedules.isEmpty() || complementDependentMode == ComplementDependentMode.OFF_MODE) {
+                            logger.info("process code: {} complement dependent in off mode or schedule's size is 0, skip "
+                                    + "dependent complement data", command.getProcessDefinitionCode());
+                        } else {
+                            dependentProcessDefinitionCreateCount += createComplementDependentCommand(schedules, command);
+                        }
                     }
                 }
                 break;
             }
             default:
                 break;
         }
-        logger.info("create complement command count: {}", createCount);
+        logger.info("create complement command count: {}, create dependent complement command count: {}", createCount
+                , dependentProcessDefinitionCreateCount);
         return createCount;
     }
+
+    /**
+     * create complement dependent command
+     */
+    @SuppressWarnings("checkstyle:OperatorWrap")
+    private int createComplementDependentCommand(List<Schedule> schedules, Command command) {
+        int dependentProcessDefinitionCreateCount = 0;
+        Command dependentCommand;
+
+        try {
+            dependentCommand = (Command) BeanUtils.cloneBean(command);
```

Comment:
Should dependentCommand set id = 0?

Reply:
I'll remove it.

Reply:
I don't think you have to worry about that.  Command.id is @TableId(value = "id", type = IdType.AUTO).